tests/callbacks_test.cpp

```diff
@@ -572,7 +572,7 @@ TEST_CASE("ProgressBarCallbackNoMaxIterationsTest", "[CallbacksTest]")
   SGDTestFunction f;
   arma::mat coordinates = f.GetInitialPoint();
 
-  StandardSGD s(0.0003, 1, 0, INT_MAX, true);
+  StandardSGD s(0.0003, 1, 0, DBL_MAX, true);
```

Comment:
Was about to comment on this one, `DBL_MAX` makes more sense here.

Reply:
Hi  @zoq I have changed `INT_MAX` to `DBL_MAX`. Kindly have a look at this whenever you have a chance.

Reply:
Yes, I realised my mistake after making the commit but changed immediately afterwards. So, @zoq @rcurtin is this now ready to get merged since all the necessary changes have been made and the necessary tests have been added? Thanks

Reply:
What's the goal here?  This sets a tolerance of `DBL_MAX`, which means that this will probably converge within a single iteration.  I suppose that's still a valid test though, since it checks if we get any output at all, and that's all we're aiming to do in the tests.